src/Storages/ObjectStorageQueue/ObjectStorageQueueIFileMetadata.cpp

```diff
@@ -344,20 +345,25 @@ void ObjectStorageQueueIFileMetadata::setFailedNonRetriable()
         return;
     }
 
-    if (Coordination::isHardwareError(responses[0]->error))
-    {
-        LOG_WARNING(log, "Cannot set file as failed: lost connection to keeper");
-        return;
-    }
+    if (Coordination::isHardwareError(code))
+        throw zkutil::KeeperMultiException(code, requests, responses);
 
-    if (responses[0]->error == Coordination::Error::ZNODEEXISTS)
+    if (responses[1]->error == Coordination::Error::ZNONODE)
```

Comment:
This one is a new check, which is added to avoid logical error thrown below, because this error is normal in case expired keeper session.